src/libsyntax/parse/parser.rs

```diff
@@ -5940,6 +5940,14 @@ impl<'a> Parser<'a> {
             self.expect(&token::OpenDelim(token::Brace))?;
             let mut trait_items = vec![];
             while !self.eat(&token::CloseDelim(token::Brace)) {
+                if let token::DocComment(_) = self.token {
+                    if self.look_ahead(1, |tok| tok == &token::Token::CloseDelim(token::Brace)) {
+                        self.span_fatal_err(self.span, Error::UselessDocComment).emit();
```

Comment:
It'd be nice if you used `struct_span_err(...).emit()` instead, and then consume the doc (as you're doing here with `self.bump()`. That way the parser will "succeed" and continue to type checking in the face of this.

Reply:
I tried doing that, but I was unable to find a way to emit an `Error` without using `span_fatal_err`. All other methods of `Parser`currently use a `&str` instead. I might look deeper into the diagnostics system once I have the time. Might take a while though :cry: